pages/glossary/start-end-content.md

```diff
@@ -0,0 +1,15 @@
+---
+title: Start and end of the content
+key: start-end-content
+unambiguous: true
+objective: true
+---
+
+Given a set of nodes S and a node N:
+
+- N is _at the start_ of S if N is located (in tree order) before the first [perceivable content][] in S (included), and after the last [perceivable content][] which is before all [perceivable content][] in S (excluded).
+- N is _at the end_ of S if N is located (in tree order) after the last [perceivable content][] in S (excluded), and before the first [perceivable content][] which is after all [perceivable content][] in S (included).
```

Comment:
I'm not understanding what you mean with (included) and (excluded) in this definition, because you don't say anywhere that N belongs to S.
If N does not belong to S, what is N excluded from?

Reply:
That's related to whether "before the first perceivable content in S" (which is a set) includes or excludes "the first perceivable content in S".
I.e. "the set of nodes before X" => is X included or excluded?
"N is located before [X=the first perceivable content in S]" => can N be X or not?

Consider 0, **1**, 2, {3, **4**, 5, **6**, 7}, 8, **9** with S in curly brackets and perceivable content in bold (1, 4, 6, 9).
The idea of start and end is for jumping before or after S and so the goal is that we hide off screen (by scrolling) exactly enough of the perceivable content (it doesn't matter whether we hide the decorative non-perceivable content).

The first perceivable content in S is 4, the last before S is 1.
Thing that are "before the first perceivable content in S (included) are 0, 1, 2, 3, and 4 (**included** that's what I,m trying to say).
Things that are "after the last perceivable before S (excluded)" are 2, 3, …, 9. 1 is not part of it because it is excluded.
That is, I want the segments […; 4] (4 included) and ]1; …] (1 excluded).

So, if I want to move "at the start of S", i.e. have scrolling hide stuff before S, but nothing inside S, then I can scroll so that the first visible stuff is either 2, 3, or 4. All of these are "before the first perceivable content in S (included)" and "after the last perceivable content before S (excluded)".
1 is not "after the last perceivable before S (excluded)". 5 is not "before the first perceivable in S (included)".
Jumping to either 2, 3 or 4 is good for "at the start of S" since I will be able to read all the important stuff in S, without reading anything before S (1) nor skipping anything from S.
Similarly, to jump after S I have to go to either 7, 8 or 9 so that I skip everything that was in S and nothing that is after. So "before 9 (included) and "after 6 (excluded)".

I guess this might need a less "formal logic" formulation 🤓 Maybe just adding the example could help understand this… Ideas?

Reply:
Can't exactly tell you why, but today, even before reading your thorough explanation, this all made sense. Still, if before I had doubts, then it probably needs improving. 

One suggestion that I have is to make clear that both "at the start" and "at the end" can be sets of nodes (instead of a single node). Then a node would be "at the start" of S if it belongs to that set. This is what the note at the end is all about, but perhaps if it came earlier might help. What do you think?

Reply:
Rephrased the defs, with the new deh of "block of content", it seems to work out better, I think…